dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -0,0 +1,519 @@
+
+    /**
+     * Finds the action associated to the stepId
+     * @param stepId String
+     * @param user   User
+     * @return List of WorkflowAction
+     */
+    public List<WorkflowAction> findActions(final String stepId, final User user) {
+
+        WorkflowStep workflowStep    = null;
+        List<WorkflowAction> actions = null;
+
+        try {
+
+            Logger.debug(this, "Looking for the stepId: " + stepId);
+            workflowStep = this.workflowAPI.findStep(stepId);
+        } catch (Exception e) {
+
+            Logger.error(this, e.getMessage(), e);
+        }
+
+        if (null != workflowStep) {
+
+            try {
+
+                Logger.debug(this, "Looking for the actions associated to the step: " + stepId);
+                actions = this.workflowAPI.findActions(workflowStep, user);
+            } catch (DotDataException  | DotSecurityException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotWorkflowException(e.getMessage(), e);
+            }
+        } else {
+            throw new DoesNotExistException("Workflow-does-not-exists-step");
+        }
+
+        return (null == actions)? Collections.emptyList(): actions;
+    } // findActions.
+
+    /**
+     * Finds the steps by schemeId
+     * @param schemeId String
+     * @return List of WorkflowStep
+     */
+    public List<WorkflowStep> findSteps(final String schemeId) {
+
+
+        List<WorkflowStep> workflowSteps  = null;
+        WorkflowScheme     workflowScheme = null;
+
+        try {
+
+            Logger.debug(this, "Looking for the schemeId: " + schemeId);
+            workflowScheme = this.workflowAPI.findScheme(schemeId);
+        } catch (DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+        }
+
+        if (null != workflowScheme) {
+
+            try {
+
+                workflowSteps = this.workflowAPI.findSteps(workflowScheme);
+            } catch (DotDataException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotWorkflowException(e.getMessage(), e);
+            }
+        } else {
+
+            throw new DoesNotExistException("Workflow-does-not-exists-scheme");
+        }
+
+        return workflowSteps;
+    } // findSteps.
+
+    /**
+     * Returns if the action associated to the actionId parameter is new and clone the retrieved action into a new one.
+     * @param actionId String
+     * @return IsNewAndCloneItResult
+     */
+    protected IsNewAndCloneItResult isNewAndCloneIt (final String actionId) {
+
+
+        final WorkflowAction newAction = new WorkflowAction();
+        boolean isNew = true;
+
+        try {
+
+            final WorkflowAction origAction = this.workflowAPI.findAction
+                    (actionId, APILocator.getUserAPI().getSystemUser());
+            BeanUtils.copyProperties(newAction, origAction);
+            isNew = !(origAction !=null || !origAction.isNew());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "origAction" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)